blogposts/2021/release-3.29.md

```diff
@@ -0,0 +1,118 @@
+    url: https://github.com/sourcegraph/sourcegraph/pull/21522
+    category: Search
+  - description: "Sourcegraph will now refuse to start if there are unfinished [out-of-band-migrations](https://docs.sourcegraph.com/admin/migrations) that are deprecated in the current version. See the [upgrade documentation](https://docs.sourcegraph.com/admin/updates) for changes to the upgrade process."
+    url: https://github.com/sourcegraph/sourcegraph/pull/20967
+    category: Admin
+# Fixed
+  - description: "Stricter validation of structural search queries. The `type:` parameter is not supported for structural searches and returns an appropriate alert."
+    url: https://github.com/sourcegraph/sourcegraph/pull/21487
+    category: Search
+  - description: Batch changeset specs that are not attached to changesets will no longer prematurely expire before the batch specs that they are associated with.
+    url: https://github.com/sourcegraph/sourcegraph/pull/21678
+    category: Batch Changes
+  - description: Code insights line chart no longer has a negative quadrant
+    url: https://github.com/sourcegraph/sourcegraph/pull/22018
+    category: 
+  - description: "Correctly handle field aliases in the query (like `r:` versus `repo:`) when used with `contains` predicates."
+    url: https://github.com/sourcegraph/sourcegraph/pull/22105
+    category: Repositories
+# Removed
+  - description: "The deprecated GraphQL `icon` field on CommitSearchResult and Repository was removed."
+    url: https://github.com/sourcegraph/sourcegraph/pull/21310
+    category: Search
+  - description: "The undocumented `index` filter was removed from search type-ahead suggestions."
+    url: https://github.com/sourcegraph/sourcegraph/issues/18806
+    category: Search
+  - description: "Code host connection tokens aren't used for creating changesets anymore when the user is site admin and no credential has been specified."
+    url: https://github.com/sourcegraph/sourcegraph/issues/16814
+    category: Admin
+
+---
+
+Sourcegraph 3.29 is now available! For this release, we introduced search results ranking, made it easier to create Code Insights, and added support for bulk actions with Batch Changes.
+
+## Search results ranking
+We're making search results more relevant, starting with better prioritization of repos vs. code in your search results. Search results and suggestions are now ranked by descending star count on sourcegraph.com and customer instances (for repos with stars), making it easier to find specific repos. Start searching on sourcegraph.com to experience the new ranking.
+
+Star count import from github.com/sourcegraph/ghdump is done. We now have 418321 repos with star data in the production repo table.
+Suggestions and search results are now ranked by descending star count on sourcegraph.com (and customer instances if repos there have stars). Try it out live!
```

Comment:
I think this line was maybe meant to be deleted when the reworded paragraph above was inserted? Appears to be repetition